packages/web_ui/src/components/InstanceViewPage.jsx

```diff
@@ -83,7 +83,11 @@ function formatOutputColored(output, key) {
 	let info = "";
 	if (output.type === "log") {
 		let level = output.level;
-		if (level === "Info") {
+		if (level === "Script") {
```

Comment:
The levels already are used twice in the code. One typo and you are racking your brain why something does not work for hours. Let's just say I have wasted A LOT of time to this specific error.... :D

Reply:
I get your sentiment here.  I've banged my head on similar issues before.  But I don't think putting it as constants is going to help much.  You see in JavaScript everything is objects, so the way it would have to be exposed would be through properties of an object.  Like for example

```js
const levels = {
    INFO: "Info",
    SCRIPT: "Script,
};
```

And then you'd access it like `libFactorio.levels.INFO`, or some appropriate place.  The problem is that accessing a property that doesn't exist is not an error, it just returns `undefined`.  So if you fat finger it like `if (level === libFactorio.levels.IINFO)` it wouldn't become an error, it would become equivalent to `if (level === undefined)`  which works more or less just the same as `if (level === "IInfo")` does.

If we were using TypeScript we could easily leverage the type system to prevent these kinds of mistakes, and I'd totally be for taking advantage of such things.  But we're not, so for now I would say it's another reason to use TypeScript.

Reply:
The idea here is that this would be caught on the syntax highlighting and by code analysis tooling. 
And code auto-complete makes it easier to not make a mistake. 

It's true - for runtime, it will make no difference if it's one or the other type of error. We are talking about human here :)

Reply:
I don't think it makes much of a difference for him if he still uses Vim.